python/foglamp/services/core/api/service.py

```diff
@@ -95,42 +96,51 @@ def get_service_records():
+        try:
+            # "plugin_module_path" is fixed by design. It is MANDATORY to keep the plugin in the exactly similar named
+            # folder, within the plugin_module_path.
+            plugin_module_path = "foglamp.plugins.south" if service_type == 'south' else \
+                                 "foglamp.plugins.north" if service_type == 'north' else None
+            import_file_name = "{path}.{dir}.{file}".format(path=plugin_module_path, dir=plugin, file=plugin)
+            _plugin = __import__(import_file_name, fromlist=[''])
+        except ImportError as ex:
+            raise web.HTTPInternalServerError(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
+        except Exception as ex:
+            raise web.HTTPInternalServerError(reason='Failed to create plugin configuration. {}'.format(str(ex)))
+
+        storage = connect.get_storage()
 
         # Check that the process name is not already registered
         payload = PayloadBuilder().SELECT("name").WHERE(['name', '=', name]).payload()
-        result = await storage.query_tbl_with_payload('scheduled_processes', payload)
+        result = storage.query_tbl_with_payload('scheduled_processes', payload)
         count = result['count']
         if count != 0:
             raise web.HTTPBadRequest(reason='A service with that name already exists')
 
         # Check that the schedule name is not already registered
         payload = PayloadBuilder().SELECT("schedule_name").WHERE(['schedule_name', '=', name]).payload()
-        result = await storage.query_tbl_with_payload('schedules', payload)
+        result = storage.query_tbl_with_payload('schedules', payload)
         count = result['count']
         if count != 0:
             raise web.HTTPBadRequest(reason='A schedule with that name already exists')
+        # ----------------------------------------
 
-        # First create the scheduled process entry for our new service
-        if service_type == 'south':
-            script = '["services/south"]'
-            plugin_module_path = "foglamp.plugins.south"
-        if service_type == 'north':
-            script = '["services/north"]'
-            plugin_module_path = "foglamp.plugins.north"
+        # Now first create the scheduled process entry for the new service
+        script = '["services/south"]' if service_type == 'south' else \
+                 '["services/north"]' if service_type == 'north' else None
         payload = PayloadBuilder().INSERT(name=name, script=script).payload()
         try:
-            res = await storage.insert_into_tbl("scheduled_processes", payload)
+            res = storage.insert_into_tbl("scheduled_processes", payload)
+        except StorageServerError as ex:
+            err_response = ex.error
+            raise web.HTTPInternalServerError(reason='Failed to created scheduled process. {}'.format(err_response))
         except Exception as ins_ex:
```

Comment:
please move as parent as we have  this at multiple places; and actually on generic exception aiohttp should capture/ return the same way even if you don't define. but yes for logging purpose keeping it as parent may be a valid idea!